templates/tomcat/login.config.erb

```diff
@@ -0,0 +1,13 @@
+CertificateLogin {
```

Comment:
For untemplated files I prefer them to live in `files/` and use `file()` in the puppet code

Reply:
Can you paste the syntax for that? I find files to be confusing compared to templates. Templates brings uniformity to the codebase, and is cleaner to understand than the file syntax IMO. I get that a template should be "templated" but I prefer the fact that uniformity using just templates brings can simplify the understanding and code within the codebase.

Reply:
```puppet
file { 'myfile':
  content => file('candlepin/tomcat/login.config'),
}
```

Then `git mv templates/tomcat/login.config.erb files/tomcat/login.config`

IMHO this isn't really that different from a template and very explicit that there's no template magic going on.

Reply:
This brings back memories now. That sytax is nice and clean and I like it. For our tests to pass we cannot use it though and fall back to:

```
    source  => "puppet:///modules/candlepin/tomcat/jaas.conf",
```

Which is the part I always found ugly and inconsistent. I've made the changes already locally. But I did want to point that out.

Reply:
Yes, the `file()` method is newer (Puppet 4?). It's part of the catalog (compiled on the master) while with `source => ` the agent reaches out to to a resource. That can be something on the agent's local filesystem but also any http(s):// resource. The `puppet://` part is a shorthand for the Puppet fileserver. Other than the style it also has the downside that it's a separate HTTP request with its own overhead. For small files (IIRC < 4k) including in the catalog is actually faster as well.